api/v1alpha1/nodepool_types.go

```diff
@@ -701,6 +701,18 @@ type Volume struct {
 	//
 	// +optional
 	IOPS int64 `json:"iops,omitempty"`
+
+	// Encrypted is whether the volume should be encrypted or not.
+	// +optional
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="Encrypted is immutable"
+	Encrypted *bool `json:"encrypted,omitempty"`
+
+	// EncryptionKey is the KMS key to use to encrypt the volume. Can be either a KMS key ID or ARN.
+	// If Encrypted is set and this is omitted, the default AWS key will be used.
+	// The key must already exist and be accessible by the controller.
+	// +optional
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="EncryptionKey is immutable"
+	EncryptionKey string `json:"encryptionKey,omitempty"`
```

Comment:
I was thinking this one we might want to keep it mutable, e.g. hey gets compromised.